### PR TITLE
[WFLY-19453] Minimize size of Otel API module

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-19453

Introduce new modules for shared artifacts
Update dependencies as needed

### CODE REVIEW

File: galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml [hunk 1]
```diff
@@ -9,16 +9,14 @@
...
-        <artifact name="${io.opentelemetry:opentelemetry-api-events}"/>
-        <artifact name="${io.opentelemetry:opentelemetry-semconv}"/>
-        <artifact name="${io.opentelemetry.instrumentation:opentelemetry-instrumentation-api}"/>
-        <artifact name="${io.opentelemetry.instrumentation:opentelemetry-instrumentation-api-semconv}"/>
-        <artifact name="${io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations}"/>
-        <artifact name="${io.opentelemetry.instrumentation:opentelemetry-instrumentation-annotations-support}"/>
     </resources>
 
     <dependencies>
+        <module name="io.opentelemetry.api.events" export="true"/>
```
Comment (bstansberry) [inline galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml:15]: Why export? Why not just add the dep to the deployment via the DUP?

File: galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml [hunk 2]
```diff
@@ -9,16 +9,14 @@
...
         <module name="io.opentelemetry.context"/>
+        <module name="io.opentelemetry.instrumentation.annotations"/>
+        <module name="io.opentelemetry.instrumentation.api"/>
+        <module name="io.opentelemetry.semconv" export="true"/>
```
Comment (bstansberry) [inline galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml:19]: Same comment as above about exporting.

File: galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/smallrye/opentelemetry/main/module.xml [hunk 3]
```diff
@@ -16,9 +16,14 @@
 
     <dependencies>
         <module name="io.opentelemetry.api"/>
+        <module name="io.opentelemetry.api.events"/>
         <module name="io.opentelemetry.context"/>
-        <module name="io.opentelemetry.exporter"/>
+        <module name="io.opentelemetry.exporter" export="true"/>
+        <module name="io.opentelemetry.otlp" export="true"/>
```
Comment (bstansberry) [inline galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/smallrye/opentelemetry/main/module.xml:22]: Same comment as above re exporting.

File: galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml (hunk 1 above)
Comment (jasondlee) [reply galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml]: Updated.

File: galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml (hunk 2 above)
Comment (jasondlee) [reply galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml]: Updated.

File: galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/smallrye/opentelemetry/main/module.xml (hunk 3 above)
Comment (jasondlee) [reply galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/smallrye/opentelemetry/main/module.xml]: Updated.

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @jasondlee

**Pull request merged**